File: typescript/base-hmi-types/src/images/HmiImage.ts
import { IHmiObject } from "../IHmiObject.js";
import { HmiImageType } from "./HmiImageType.js";

export class HmiImage implements IHmiObject {
  id?: string;
  name?: string;
  lastModified?: Date;
  imageType: HmiImageType = HmiImageType.Unknown;
  mimeType?: string;
  data = new Uint8Array();
}

File: src/BaseHmiTypes/Images/HmiImage.cs
namespace BaseHmiTypes.Images;

public sealed class HmiImage : IHmiObject
{
    public string? Id { get; set; }

    public string? Name { get; set; }

    public DateTime? LastModified { get; set; }

    public HmiImageType ImageType { get; set; } = HmiImageType.Unknown;

    public string? MimeType { get; set; }

    public byte[] Data { get; set; } = Array.Empty<byte>();
}
